src/test/java/io/jenkins/plugins/coverage/model/QualityGateEvaluatorTest.java

```diff
@@ -0,0 +1,157 @@
+        sut.add(0, QualityGate.QualityGateType.INSTRUCTION, QualityGate.QualityGateResult.UNSTABLE);
+        sut.add(0, QualityGate.QualityGateType.LINE, QualityGate.QualityGateResult.UNSTABLE);
+        sut.add(0, QualityGate.QualityGateType.BRANCH, QualityGate.QualityGateResult.UNSTABLE);
+        sut.add(0, QualityGate.QualityGateType.PACKAGE, QualityGate.QualityGateResult.UNSTABLE);
+        sut.add(0, QualityGate.QualityGateType.CLASS, QualityGate.QualityGateResult.UNSTABLE);
+
+        QualityGateStatus result = sut.evaluate(stats, logger);
+
+        assertThat(result).isEqualTo(QualityGateStatus.PASSED);
+    }
+
+    @Test
+    void evaluateWarning() {
+        Logger logger = new Logger();
+        CoverageNode tree = readExampleReport();
+        CoverageStatistics stats = new CoverageStatistics(tree.getMetricsDistribution());
+
+        QualityGateEvaluator sut = new QualityGateEvaluator();
+        sut.add(0.9, QualityGate.QualityGateType.FILE, QualityGate.QualityGateResult.UNSTABLE);
+
+        QualityGateStatus result = sut.evaluate(stats, logger);
+
+        assertThat(result).isEqualTo(QualityGateStatus.WARNING);
+    }
+
+    @Test
+    void evaluatePassed() {
+        Logger logger = new Logger();
+        CoverageNode tree = readExampleReport();
+        CoverageStatistics stats = new CoverageStatistics(tree.getMetricsDistribution());
+
+        QualityGateEvaluator sut = new QualityGateEvaluator();
+        sut.add(0.3, QualityGate.QualityGateType.FILE, QualityGate.QualityGateResult.UNSTABLE);
+
+        QualityGateStatus result = sut.evaluate(stats, logger);
+
+        assertThat(result).isEqualTo(QualityGateStatus.PASSED);
+    }
+
+    @Test
+    void evaluateFailed() {
+        Logger logger = new Logger();
+        CoverageNode tree = readExampleReport();
+        CoverageStatistics stats = new CoverageStatistics(tree.getMetricsDistribution());
+
+        QualityGateEvaluator sut = new QualityGateEvaluator();
+        sut.add(1, QualityGate.QualityGateType.FILE, QualityGate.QualityGateResult.FAILURE);
+
+        QualityGateStatus result = sut.evaluate(stats, logger);
+
+        assertThat(result).isEqualTo(QualityGateStatus.FAILED);
+    }
+
+    @Test
+    void shouldChangeStatus() {
+        Logger logger = new Logger();
+        CoverageNode tree = readExampleReport();
+        CoverageStatistics stats = new CoverageStatistics(tree.getMetricsDistribution());
+
+        QualityGateEvaluator sut = new QualityGateEvaluator();
+        sut.add(0.8, QualityGate.QualityGateType.FILE, QualityGate.QualityGateResult.UNSTABLE);
+        sut.add(0.5, QualityGate.QualityGateType.BRANCH, QualityGate.QualityGateResult.FAILURE);
+        sut.add(0.95, QualityGate.QualityGateType.INSTRUCTION, QualityGate.QualityGateResult.FAILURE);
+
+        QualityGateStatus result = sut.evaluate(stats, logger);
+
+        assertThat(result).isEqualTo(QualityGateStatus.FAILED);
+    }
+
```

Comment:
In order to get 100% branch coverage you need an additional test that evaluates L41 to false (e.g., 2 times an UNSTABLE quality gate that has been hit)